### ISSUE TITLE
opt,distsql: literal constants flow back and forth for every query using DISTINCT

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#48166]
tldr: the code currently distributes the text of error messages *even in the common case where there is no error*.

See: field `ErrorOnDup` in `DistinctSpec` (processors_sql.proto)

This is really an architectural misdesign in the code, which also runs against the grain of making error messages ssafely reportable in telemetry:

- the string flows across nodes even in the common case where there is no error.
- there is no API guarantee that the text is not dynamically generated and may contain PII -- so at the point the error object is constructed, we can't assume this text is reportable.

The correct API would be to send an **enum** in the processor spec, then select the type of error to return based on the enum value.

cc @RaduBerinde for triage
